src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -660,19 +661,18 @@ safe_void_coroutine WindowEmperor::_dispatchCommandlineCurrentDesktop(winrt::Ter
             if (desktopId == currentDesktop && lastActivatedTime > max)
             {
                 max = lastActivatedTime;
-                mostRecentWeak = w;
+                mostRecent = w;
             }
         }
-    }
-
-    // GetVirtualDesktopId(), as current implemented, should always return on the main thread.
-    _assertIsMainThread();
 
-    const auto mostRecent = mostRecentWeak.lock();
-    auto window = mostRecent.get();
-
-    if (!window)
+        // GetVirtualDesktopId(), as current implemented, should always return on the main thread.
+        _assertIsMainThread();
+        window = mostRecent.get();

```

Comment:
I don't actually understand how this fixes it. It seems to be the same implementation as before, but without the weak pointer...?